### DISCUSSION TITLE
Missing IMU messages after Humble switch

### DISCUSSION THREAD

**Discussion by alfred-wrigfors** [ouster-lidar/ouster-ros#242] [Q&A]
Hi!

We have recently switched from Galactic to Humble and are encountering an issue with missed frames. It is specifically the IMU frames. The we have no problem with the PointCloud2 messages, but the IMU messages are sometimes missed (about once every 2 seconds on average). We have of-cource cheched that it is not our code that runs to slow and therefore discard the messages (becouse of queuesize etc). When listening to the topics, the frames/msgs har just not sent (looking at times-stamps for messages).

We have tried with the community driver and basically not having this problem at all. We do have issues with the community driver as well, but that seems to be more of a system issue than anything else (sensor becoming unreachable via ethernet after ~2min, cable unplugging and re-plugging fixes the issued only to happen again after ~2min).

Have anybody else had this issue? Is there anything we can do to troubleshoot?

**Comment by Samahu** [comment]
Hi @alfred-wrigfors, the sensor sends data via udp protocol, which isn't reliable so if there is a significant packet drop between the sensor and the machine the ouster driver runs on you could refer to [Ouster Networking Guide](https://static.ouster.dev/sensor-docs/image_route1/image_route3/networking_guide/networking_guide.html) to better optimize your network setup. On the other hand, if the problem within ROS I would suggest that you try to set "use_system_default_qos" param file to "true" within the `driver_params.yaml`.
